usersession/agent/rest_api.go

```diff
@@ -334,3 +348,58 @@ func postRefreshFinishedNotification(c *Command, r *http.Request) Response {
 	}
 	return SyncResponse(nil)
 }
+
+func postBeginDeferredRefreshNotification(c *Command, r *http.Request) Response {
+	if ok, resp := validateJSONRequest(r); !ok {
+		return resp
+	}
+
+	decoder := json.NewDecoder(r.Body)
+
+	// BeginDeferredRefreshNotificationInfo holds information about pending snap refresh provided by snapd.
+	var refreshInfo userclient.BeginDeferredRefreshNotificationInfo
+	if err := decoder.Decode(&refreshInfo); err != nil {
+		return BadRequest("cannot decode request body into pending snap deferred refresh info: %v", err)
+	}
+
+	go monitorChanges(refreshInfo, c.s.notificationMgr)
+	return SyncResponse(&resp{
+		Type:   ResponseTypeError,
+		Status: 200,
+		Result: &errorResult{
+			Message: "Ok",
+		},
+	})
+}
+
+// This function sends a standard desktop notification using the DBus API for notifications.
```

Comment:
It's our code style to start docstrings with their names
```suggestion
// sendDesktopStandardNotification sends a standard desktop notification using the DBus API for notifications.
```